In [1]:
import pandas as pd
import numpy as np
import math
import matplotlib.pyplot as plt
import seaborn as sns
import scipy.stats as ss

Please answer questions 1, 5, 7, 8, 9, 10, 13, 14, 17, and 18.

### 1

The riskless rate of return is 4% and the expected return on the market portfolio is 8%.

 

What is the expected return on the shares of a firm with a beta of 1.15 ?

$$
    CAPM = R_{riskfree} + {Beta}*[E(R_{m}) - R_{riskfree}]
$$

In [2]:
4 + 1.15 * (8-4)

8.6

### 5

The market is expected to return 7% and has a volatility of 16%. The riskless rate of return is 3%. An investor with \\$60,000 is willing to bear a maximum volatility of 20% on his portfolio.

 

How should he allocate his \\$60,000 investment between the market and the riskless asset ?

In [4]:
def vol_rf_market(weight1, weight2, return1, return2, vol1, vol2, corr):
    
    combined_var = weight1**2 * vol1**2 + weight2**2 * vol2**2 + 2*weight1*weight2*vol1*vol2*corr
    
    return combined_var**0.5

In [10]:
vol_rf_market(.5, .5, 0.07, 0.03, 0.16, 0.0, 0.0)

0.08

In [15]:
w1 = 1

for i in range(10):
    w2 = 1 - w1
    print(vol_rf_market(w1, w2, 0.07, 0.03, 0.16, 0, 0), round(w1, 3), round(w2, 3))
    
    w1 += 0.05

0.16 1 0
0.168 1.05 -0.05
0.17600000000000002 1.1 -0.1
0.18400000000000002 1.15 -0.15
0.19200000000000003 1.2 -0.2
0.20000000000000007 1.25 -0.25
0.20800000000000005 1.3 -0.3
0.21600000000000005 1.35 -0.35
0.22400000000000006 1.4 -0.4
0.2320000000000001 1.45 -0.45


### 7

An asset has a beta coefficient of 1.10, a volatility of 24% and a correlation of 0.825 with the market portfolio.

 

What is the market volatility ?



$$
    BETA = \frac{Corr(R_i, R_m) * Vol_i}{Vol_m}
$$

In [16]:
0.825 * .24

0.19799999999999998

In [18]:
0.198/1.1

0.18

### 8

Asset A has a volatility of 20% and a beta of 0.95. Asset B has a lower volatility of 15% and a beta of 1.10.


In a portfolio context with many assets, which of the following statements is correct ?

In [20]:
# Asset A is riskier than asset B.

# Asset B is riskier than asset A.

# The answer depends on which measure of risk is chosen.

# There is an inconsistency in the information provided.

In [21]:
# (corrA * .2)/.95 = (corrB * .15)/1.1

In [31]:
# 1100corrA * 220 = 950corrB * 142.5

In [32]:
1100 * 220

242000

In [33]:
950*142.5

135375.0

In [34]:
242000 / 135375

1.7876269621421976

### 9 

Asset A has an expected return of 7.80% with 20% volatility. Asset B has a higher expected return of 8.40% with 15% volatility.

 

Which of the following statements is correct ?

In [37]:
# The data is inconsistent with the CAPM because the ration of expected returns (7.80%/8.40%) 
# is different from the ratio of their corresponding volatilities (20%/15%)


# The data is inconsistent with the CAPM that posits a positive relationship between expected 
# returns and risk: higher expected returns should be associated with higher risk, not lower risk.


# The data is consistent with the CAPM only if investors display a favorable attitude towards risk.


# The data does not necessarily contradict the CAPM because the relevant measure of the risk of an 
# asset, according to the CAPM is beta not volatility.

### 10

Portfolio P has an expected return of 14% and a volatility of 32%. The expected market risk premium and the riskless rate are both 5%. The market volatility is 16%.

 

Is portfolio P efficient according the the CAPM?

In [38]:
(5/16) * 32 + 5

15.0

In [40]:
# Not efficient, should have returned 15% rather than 14%

### 13

What is the cost of equity capital of a company with a beta of 1.16 ? (The expected excess return on the market portfolio is 4% and the riskless rate of return is 3%.)

In [41]:
3 + 1.16*4

7.64

### 14

You want to estimate a firm's cost of equity. The firm's shares have a volatility of 15% and a correlation of 0.80 with the market. You know that there is an efficient portfolio with a volatility of 20% and an expected return of 15%. The expected return on the market portfolio is 10% and the riskless rate of return is 5%.

In [42]:
# calculate beta

(0.8 * .15)/.2

0.6

In [46]:
0.6 * (15-5) + 5

11.0

### 17  

Manager A has achieved a 12% return with a volatility of 20% and a beta of 1.10. Manager B has achieved a 13% return with a volatility of 25% and a beta of 1.20. Over the same period the market portfolio yielded 11% and the riskless rate was 6%.

 

Which of the following statements is correct ?

In [47]:
# Manager A underperformed the market and Manager B overperformed it.


# Manager A outperformed the market and Manager B underperformed it.


# Both managers underperformed the market.


# Both managers overperformed the market.

In [48]:
1.1 * (11-6) + 6

11.5

In [49]:
1.2 * (11-6) + 6

12.0

### 18  

Manager A has achieved a 11% return with a volatility of 30% and a beta of 1.20. Manager B has achieved a 13% return with a volatility of 25% and a beta of 0.90. Over the same period, the market portfolio yielded 12% and a riskless rate was 7%.

 

Which of the following statements is correct ?



Manager A produced an alpha of 1.5% and Manager B an alpha of -2%.


Manager A produced an alpha of 2% and Manager B an alpha of -1.5%.


Manager A produced an alpha of -1.5% and Manager B an alpha of 2%.


Manager A produced an alpha of -2% and Manager B an alpha of 1.5%.

In [54]:
1.2*(12-7) + 7

13.0

In [53]:
.9*(12-7) + 7

11.5

In [56]:
(11-13)

-2

In [57]:
13-11.5

1.5